### PR TITLE
8301004: httpclient: Add more debug to HttpResponseInputStream

### PR DESCRIPTION
**Pull request by dfuch**
The AsyncExecutorShutdown test has been observed failing (rarely) in our CI. The root cause seems to be an assertion error in HttpResponseInputStream. More debug traces could help with diagnosing how we reached this assertion.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301004](https://bugs.openjdk.org/browse/JDK-8301004): httpclient: Add more debug to HttpResponseInputStream

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12169/head:pull/12169` \
`$ git checkout pull/12169`

Update a local copy of the PR: \
`$ git checkout pull/12169` \
`$ git pull https://git.openjdk.org/jdk pull/12169/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12169`

View PR using the GUI difftool: \
`$ git pr show -t 12169`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12169.diff">https://git.openjdk.org/jdk/pull/12169.diff</a>

</details>

### CODE REVIEW

File: src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java [hunk 1]
```diff
@@ -557,10 +559,14 @@ public void onSubscribe(Flow.Subscription s) {
                         closed = this.closed;
                         if (!closed) {
                             this.subscription = s;
-                            assert buffers.remainingCapacity() > 1; // should contain at least 2
+                            // should contain at least 2
+                            assert buffers.remainingCapacity() > 1
+                                    : "buffers capacity: " + buffers.remainingCapacity()
+                                    + "closed: " + closed + "failed: " + failed;
```
Comment (jaikiran) [inline src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java:565]: Hello Daniel, I think a space before `closed:` and `failed:` would be good. Other than that, this looks good.

**Review by jaikiran — APPROVED** [review]

**Review by jaikiran — APPROVED** [review]

File: src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java (hunk 1 above)
Comment (dfuch) [inline src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java:565]:
```suggestion
                                    + " closed: " + closed + "failed: " + failed;
```

File: src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java [hunk 2]
```diff
@@ -557,10 +559,14 @@ public void onSubscribe(Flow.Subscription s) {
                         closed = this.closed;
                         if (!closed) {
                             this.subscription = s;
-                            assert buffers.remainingCapacity() > 1; // should contain at least 2
+                            // should contain at least 2
+                            assert buffers.remainingCapacity() > 1
+                                    : "buffers capacity: " + buffers.remainingCapacity()
+                                    + " closed: " + closed + "failed: " + failed;
```
Comment (dfuch) [inline src/java.net.http/share/classes/jdk/internal/net/http/ResponseSubscribers.java:565]:
```suggestion
                                    + " closed: " + closed + " failed: " + failed;
```

**Review by jaikiran — APPROVED** [review]

**Pull request closed without merging**